v3/src/components/graph/components/binneddotplotdots.tsx

```diff
@@ -151,23 +236,52 @@ export const BinnedDotPlotDots = observer(function BinnedDotPlotDots(props: Plot
       pointDisplayType, anchor: circleAnchor
     })
   },
-  [graphModel, dataConfig, layout, primaryAttrRole, secondaryAttrRole, dataset, pixiPoints,
-    primaryIsBottom, pointColor, pointStrokeColor, isAnimating, pointDisplayType])
+  [dataConfig, primaryIsBottom, graphModel, primaryPlace, layout, primaryAttrRole, secondaryAttrRole,
+   drawBinBoundaries, dataset, primaryAxisScale, pixiPoints, pointColor, pointStrokeColor, isAnimating,
+   pointDisplayType, addBinBoundaryDragHandlers])
 
   usePlotResponders({pixiPoints, refreshPointPositions, refreshPointSelection})
 
-  // respond to binAlignment and binWidth changes because the axis may need to be updated
+  // Respond to binAlignment and binWidth changes because the axis may need to be updated.
   useEffect(function respondToGraphBinSettings() {
     return mstReaction(
-      () => [graphModel.binAlignment, graphModel.binWidth],
+      () => [graphModel._binAlignment, graphModel._binWidth, graphModel.binAlignment, graphModel.binWidth],
```

Comment:
I found the underscore properties were necessary. When I remove them, calling the graph model's `endBinBoundaryDrag` method doesn't have the desired effect of removing the drag classes from the boundaries and reapplying the drag-handling methods to them. I think that's because the non-underscore versions' values don't appear to change? In other words, when you've dragged a boundary and the `binWidth` is at say `5` and you stop dragging, `binWidth` will still be `5` and therefore won't trigger the reaction. We have to watch `_binWidth` for the actual change we're concerned with.  I may be wrong about that and/or there may be a more elegant way of handling that issue, though.

Reply:
Thinking about this a bit more, I should at least add something about the above to the comment for this `useEffect`, or maybe make a separate `useEffect` specific to the underscore properties? 

I think ideally we wouldn't need to check for changes to those properties, and could somehow else get `refreshPointPositions` to run at the end of a drag. That function can't really be called from `handleDragBinBoundaryEnd` though because the latter is defined first.